drools-game-engine-horserace-kjar/src/test/java/org/drools/game/horserace/tests/GameRulesTest.java

```diff
@@ -0,0 +1,123 @@
+
+package org.drools.game.horserace.tests;
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+import javax.inject.Inject;
+import org.drools.game.core.GameCallbackServiceImpl;
+import org.drools.game.core.GameMessageServiceImpl;
+import org.drools.game.core.api.GameMessageService;
+import org.drools.game.horserace.model.Room;
+import org.drools.game.model.api.Player;
+import org.drools.game.model.impl.base.BasePlayerImpl;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.api.KieBase;
+import org.kie.api.cdi.KBase;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.rule.FactHandle;
+
+@RunWith( Arquillian.class )
+public class GameRulesTest {
```

Comment:
Overall this test looks really good, but I encourage you to also create the GameAPITest as in the Capture the Flag and Escape the Room packs, which demonstrate how to use the GameSession API plus it pushes you to encapsulate into a command the actions of the player. (in this case hitting a checkpoint, will be a command)